src/odemis/gui/cont/acquisition/cryo_z_localization.py

```diff
@@ -132,28 +361,118 @@ def _cmb_stig_angle_set(self, value):
         else:
             logging.warning("Combobox stigmator angle has no value %s", value)
 
+    def _cmb_fiducial_size_get(self) -> float:
+        """
+        Get the current fiducial size based on the dropdown selection.
+        :return: size in meters
+        """
+        i = self._panel.cmb_fiducial_size.GetSelection()
+        if i == wx.NOT_FOUND:
+            logging.warning("cmb_fiducial_size has unknown value.")
+            return
+        size = self._panel.cmb_fiducial_size.GetClientData(i)
+        return size
+
+    def _cmb_fiducial_size_set(self, value: float) -> None:
+        """
+        Set the current fiducial size based on the dropdown selection.
+        :param value: Value to set in meters
+        """
+        ctrl = self._panel.cmb_fiducial_size
+        for i in range(ctrl.GetCount()):
+            d = ctrl.GetClientData(i)
+            if d == value:
+                logging.debug("Setting combobox value to %s", ctrl.Items[i])
+                ctrl.SetSelection(i)
+                break
+        else:
+            logging.warning("Combobox fiducial size has no value %s", value)
+
+    def _cmb_poi_size_get(self) -> float:
```

Comment:
The typing is incorrect. The function can return None -> `Optional[float]` (and change the `return` to `return None` to make that clearer).